In [144]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import scipy.stats as stats

In [145]:
#importing dataset into python
data=pd.read_csv('Sales_add.csv')

In [146]:
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [147]:
data.shape

(22, 5)

In [148]:
#checking for null values
data.isna().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

In [149]:
#column names
data.columns

Index(['Month', 'Region', 'Manager', 'Sales_before_digital_add(in $)',
       'Sales_After_digital_add(in $)'],
      dtype='object')

In [150]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


# 1. The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

Null hypothesis:
There is no increase in sales after stepping into digital marketing. 

Alternate hypothesis:
    There is an increase in sales after stepping into digital marketing

In [151]:
alpha=.05

In [152]:
a=data['Sales_before_digital_add(in $)']

In [153]:
b=data['Sales_After_digital_add(in $)']

In [171]:
#Calculate the t-test on TWO RELATED samples of scores, a and b.
t_value,p_value=scipy.stats.ttest_rel(a, b)

In [172]:
#one tailed p value = half of two tailed p avlue
one_tailed_p_value=float("{:.6f}".format(p_value/2))

In [173]:
print('Test statistic is %f'%float("{:.6f}".format(t_value)))

Test statistic is -12.090705


In [174]:
print('p-value for one_tailed_test is %f'%one_tailed_p_value)

p-value for one_tailed_test is 0.000000


Conclusion:


    Since p-value(=0.000000) < alpha(=0.05) We reject the null hypothesis H0.So we conclude that there is any increase in 
    sales after stepping into digital marketing

# 2.The company needs to check whether there is any dependency between the features “Region” and “Manager”.

In [158]:
data1=pd.crosstab(data['Region'],data['Manager'])
data1

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

Null hypothesis:
There is no relation between the variables. 

Alternate hypothesis:
    There is a significant relation between the two variables

In [159]:
data1.values

array([[4, 3, 3],
       [4, 1, 2],
       [1, 3, 1]], dtype=int64)

In [160]:
observed_values=data1.values

In [161]:
#finding chi-square value and p value
chi,p,dof,expected_values=stats.chi2_contingency(data1)

In [162]:
print('chi squared value = ',chi)
print('\n p value = ',p)
print('\n Degree of freedom = ',dof)
print('\n Expected values = ',expected_values)

chi squared value =  3.050566893424036

 p value =  0.5493991051158094

 Degree of freedom =  4

 Expected values =  [[4.09090909 3.18181818 2.72727273]
 [2.86363636 2.22727273 1.90909091]
 [2.04545455 1.59090909 1.36363636]]


In [165]:
#critical chi-square value
Critical_chi_squared=scipy.stats.chi2.ppf(1-alpha, dof)

In [166]:
Critical_chi_squared

9.487729036781154

Since p value (0.5493991051158094) is greater than significance factor( alpha = (0.05)), we fail to reject Null Hypothesis. There is no relationship between the 2 features 'Region' and 'Manager'

Since Chi-square value (3.050566893424036) is less less than the Critical Chi-squared value(9.487729036781154), we fail to reject Null Hypothesis. There is no relationship between the 2 features 'Region' and 'Manager'

# Another method

In [167]:
import researchpy

In [168]:
crosstab,res,exp=researchpy.crosstab(data['Region'],data['Manager'],test='chi-square',expected_freqs=True)

Result of the test:

In [169]:
res

Chi-square test  results
0  Pearson Chi-square ( 4.0) =    3.0506
1                    p-value =    0.5494
2                 Cramer's V =    0.2633

Expected values:

In [170]:
exp

Manager                        
Manager    Manager - A Manager - B Manager - C
Region                                        
Region - A    4.090909    3.181818    2.727273
Region - B    2.863636    2.227273    1.909091
Region - C    2.045455    1.590909    1.363636

Since p value (0.5493991051158094) is greater than significance factor( alpha = (0.05)), we fail to reject Null Hypothesis. There is no relationship between the 2 features 'Region' and 'Manager'

Since Chi-square value (3.050566893424036) is less less than the Critical Chi-squared value(9.487729036781154), we fail to reject Null Hypothesis. There is no relationship between the 2 features 'Region' and 'Manager'